iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/DanglingLineImpl.java

```diff
@@ -188,13 +359,31 @@ public void extendVariantArraySize(int initVariantArraySize, int number, int sou
             p0.add(p0.get(sourceIndex));
             q0.add(q0.get(sourceIndex));
         }
+        if (generation != null) {
+            generation.targetP.ensureCapacity(generation.targetP.size() + number);
+            generation.targetQ.ensureCapacity(generation.targetQ.size() + number);
+            generation.voltageRegulationOn.ensureCapacity(generation.voltageRegulationOn.size() + number);
+            generation.targetV.ensureCapacity(generation.targetV.size() + number);
+            for (int i = 0; i < number; i++) {
+                generation.targetP.add(generation.targetP.get(sourceIndex));
+                generation.targetQ.add(generation.targetQ.get(sourceIndex));
+                generation.voltageRegulationOn.add(generation.voltageRegulationOn.get(sourceIndex));
+                generation.targetV.add(generation.targetV.get(sourceIndex));
+            }
+        }
```

Comment:
Move the variant management in the generation: this is an encapsulation issue, the generation should be in charge of its data.

Reply:
Hm, this was what was done at first but I think you told me to change it? Did I miss something?